sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -256,26 +256,22 @@ void gnrc_ipv6_nib_handle_timer_event(void *ctx, uint16_t type)
 static size_t _get_l2src(kernel_pid_t iface, uint8_t *l2src,
                          size_t l2src_maxlen)
 {
-    bool try_long = false;
-    int res;
     uint16_t l2src_len;
     /* maximum address length that fits into a minimum length (8) S/TL2A
      * option */
     const uint16_t max_short_len = 6;
 
-    /* try getting source address */
-    if ((gnrc_netapi_get(iface, NETOPT_SRC_LEN, 0, &l2src_len,
-                         sizeof(l2src_len)) >= 0) &&
-        (l2src_len > max_short_len)) {
-        try_long = true;
-    }
-
-    if (try_long && ((res = gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG, 0,
-                                            l2src, l2src_maxlen)) > max_short_len)) {
+    /* get source address length */
+    int res = gnrc_netapi_get(iface, NETOPT_SRC_LEN, 0,
```

Comment:
@smlng seems good to me

@miri64 I think the change has now the same behavior as before but it would be perfect if you could double check this. (the same change is also in `gnrc_ndp2.c`).

Reply:
Why is the variable declaration moved down? IMHO (and I stress that this is not a convention of the project but my personal) it is way more readable and comprehensible if it is at the top of the block, like it was before.

None of this changes except the cast in l270 seem to be about sign-compare errors, but maybe I'm mistaken. I'm not a friend of "massive" code restructuring in separate PRs and would prefer to put everything that isn't about the sign-compare error into a separate PR.

Reply:
(btw: this is also a change that was simplified in #7925, since `gnrc_netif(2)` allows me to access the L2-address directly).

Reply:
regarding code restructuring: I agree (my bad, was carried away by the idea of optimising _just a little_ 😉 ), such should be done in separate PRs if anyway.

Lets see what's left of it after merging #7925 and rebasing this one.